shotover-proxy/src/transforms/cassandra/sink_cluster/routing_key.rs

```diff
@@ -0,0 +1,2498 @@
+use cassandra_protocol::frame::{Serialize, Version};
+use cassandra_protocol::query::QueryValues;
+use cassandra_protocol::types::value::Value;
+use cassandra_protocol::types::CIntShort;
+use cassandra_protocol::types::SHORT_LEN;
+use itertools::Itertools;
+use std::io::{Cursor, Write};
+
+pub fn calculate(
+    pk_indexes: &[i16],
+    query_values: &QueryValues,
+    version: Version,
+) -> Option<Vec<u8>> {
+    let values = match query_values {
+        QueryValues::SimpleValues(values) => values,
+        _ => panic!("handle named"),
+    };
+
+    serialize_routing_key_with_indexes(values, pk_indexes, version)
+}
+
+fn serialize_routing_key_with_indexes(
```

Comment:
This is all copied from cdrs tokio, whats the plan for upstreaming?

Reply:
It could, though it's only a couple functions and not part of the "cassandra protocol" so doesn't make sense to move it there. Definitely driver functionality.

Reply:
The way I see it is, cassandra-protocol is a collection of tools for constructing your own driver, so including things like this makes sense.
What we dont want to include is things that actually make driver level decisions, anything touching things like: tcp connections, error handling, building topologies.

We also dont want to unceremoniously dump things into cassandra-protocol and then find afterwards find the the API needs reworking.
So I am quite happy to leave it as is and cleanup later when we have a better understanding of our needs.
But lets at least add a link to the relevant cdrs-tokio lines this is from so we dont forget its origin.

Reply:
My understanding is it's an implementation of this document: 
https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v5.spec

Routing is a driver functionality.

Reply:
I think we need to consider what is the purpose of cassandra-protocol.
Its to let users implement their own driver level functionality, not literally just to implement the contents of a document labelled protocol.spec

For example we did already move the Murmur3Token there.
This feels the same to me.

Reply:
(And again, all im truly asking for here right now is a link back to the source so we remember where it came from and can deduplicate in the future)

Reply:
Ah good point. Anyway I'll add that source link